### PR TITLE
[TF] Add sections on BatchNorm, AlexNet, NiN, ResNet, and VGG and misc fixes

### PR DESCRIPTION
**Pull request by terrytangyuan**
*Description of changes:*

* Add sections on BatchNorm, AlexNet, NiN, ResNet, and VGG
* Support `resize` and remove `append_last_dim` in fashion_mnist dataset loader

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

File: chapter_convolutional-modern/batch-norm.md [hunk 1]
```diff
@@ -285,6 +285,18 @@ def batch_norm(X, gamma, beta, moving_mean, moving_var, eps, momentum):
     return Y, moving_mean, moving_var
 ```
 
+```{.python .input}
+#@tab tensorflow
+from d2l import tensorflow as d2l
+import tensorflow as tf
+
+# TODO: Check whether it's necessary to re-implement from scatch here.
```
Comment (terrytangyuan) [inline chapter_convolutional-modern/batch-norm.md:293]: @mli What do you think? I feel like it's unnecessary to re-implement from scratch since we won't be reusing it in other sections anyways.

**Comment by terrytangyuan** [comment]
@mli CI is passing now. Please review when you get a chance.
